Build failure with mtl-2.3

```
Building library for retry-0.9.2.0..
[1 of 1] Compiling Control.Retry    ( src/Control/Retry.hs, /home/simon/trusteeing/retry/retry-0.9.2.0/.dist-newstyle-trustee/4059df488dc2e13f6af48218de17e58723a471c63dfeda57bd604431ae257faf7b940fe494eac6779ef0aeb347e20df89d5c997ae419dac1eb2051003a473a0b/build/x86_64-linux/ghc-8.6.5/retry-0.9.2.0/noopt/build/Control/Retry.o )


src/Control/Retry.hs:853:12: error:
    • Variable not in scope:
        lift :: m (Maybe Int) -> StateT RetryStatus m (Maybe Int)
    • Perhaps you meant one of these:
        ‘liftA’ (imported from Control.Applicative),
        ‘liftM’ (imported from Control.Monad)
    |
853 |   delay <- lift (f stat)
    |            ^^^^
```

As a Hackage trustee I have revised v0.9.2.0: https://hackage.haskell.org/package/retry-0.9.2.0/revisions/

--- a/src/Control/Retry.hs
+++ b/src/Control/Retry.hs
@@ -97,6 +97,7 @@ import           Control.Monad
 import           Control.Monad.Catch
 import           Control.Monad.Except
 import           Control.Monad.IO.Class as MIO
+import           Control.Monad.Trans.Class as TC
 import           Control.Monad.Trans.Maybe
 import           Control.Monad.Trans.State
 import           Data.List (foldl')
@@ -850,7 +851,7 @@ retryOnError policy chk f = go defaultRetryStatus
 simulatePolicy :: Monad m => Int -> RetryPolicyM m -> m [(Int, Maybe Int)]
 simulatePolicy n (RetryPolicyM f) = flip evalStateT defaultRetryStatus $ forM [0..n] $ \i -> do
   stat <- get
-  delay <- lift (f stat)
+  delay <- TC.lift (f stat)
   put $! stat
     { rsIterNumber = i + 1
     , rsCumulativeDelay = rsCumulativeDelay stat `boundedPlus` fromMaybe 0 delay

